wsimod/nodes/nutrient_pool.py

```diff
@@ -136,7 +129,48 @@ def __init__(
             self.dissolved_organic_pool,
             self.adsorbed_inorganic_pool,
         ]
+    
+    def calculate_fraction_parameters(self):
+        ''' Update fractions of nutrients input transformed into other forms in soil
+        based on the input parameters
+        Returns:
+            (dict): fraction of manure to fast pool
+            (dict): fraction of plant residue to humus pool
+            (float): fraction of dry nitrogen deposition to fast pool
+        '''
+        fraction_manure_to_fast = {
+            x: 1 - self.fraction_manure_to_dissolved_inorganic[x]
+            for x in constants.NUTRIENTS
+        }
+        fraction_residue_to_humus = {
+            x: 1 - self.fraction_residue_to_fast[x] for x in constants.NUTRIENTS
+        }
+        fraction_dry_n_to_fast = 1 - self.fraction_dry_n_to_dissolved_inorganic
+        
+        return fraction_manure_to_fast, fraction_residue_to_humus, fraction_dry_n_to_fast
```

Comment:
Would it make sense to just store these in `self` inside the function - same as some functions earlier I think. Not sure what the motivation of passing them about is

Reply:
Agree - see commit [9641dbf](https://github.com/ImperialCollegeLondon/wsi/pull/82/commits/9641dbfe17fd7034e8a0a27d438ff38e1e54e42b)